lib/index.js

```diff
@@ -19,9 +20,10 @@ module.exports = curry['default'] = curry
 function curry (fn) {
   var self = this
   return function () {
-    var args1 = slice(arguments)
+    var args = slice(arguments)
```

Comment:
The problem is that when you pass `arguments` v8 bails on optimization:

https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments